src/interbtcIndex.ts

```diff
@@ -173,29 +174,32 @@ export function indexIssueToTypedIssue(issue: interbtcIndex.IndexIssue): Issue {
     // TODO determine collateralCurrency based on vault
     return {
         ...issue,
-        wrappedAmount: BitcoinAmount.from.Satoshi(issue.amountInterBTC),
+        wrappedAmount: BitcoinAmount.from.Satoshi(issue.wrappedAmount),
         bridgeFee: BitcoinAmount.from.Satoshi(issue.bridgeFee),
-        griefingCollateral: PolkadotAmount.from.Planck(issue.griefingCollateral) as MonetaryAmount<Currency<CollateralUnit>, CollateralUnit>,
+        // NOTE: KusamaAmount is hardcoded here, theoretically need a way to pass "ParachainCurrency"
+        // In practice though this will be deprecated before we move off Kusama, hence an extra parameter is not being added for this
+        griefingCollateral: KusamaAmount.from.KSM(issue.griefingCollateral) as MonetaryAmount<Currency<CollateralUnit>, CollateralUnit>,
         btcAmountSubmittedByUser: issue.btcAmountSubmittedByUser
             ? BitcoinAmount.from.Satoshi(issue.btcAmountSubmittedByUser)
             : undefined,
         refundAmountBTC: issue.refundAmountBTC ? BitcoinAmount.from.Satoshi(issue.refundAmountBTC) : undefined,
         executedAmountBTC: issue.executedAmountBTC ? BitcoinAmount.from.Satoshi(issue.executedAmountBTC) : undefined,
-        userParachainAddress: issue.userDOTAddress,
-        vaultParachainAddress: issue.vaultDOTAddress
+        userParachainAddress: issue.userParachainAddress,
+        vaultParachainAddress: issue.vaultParachainAddress
     };
 }
 
 export function indexRedeemToTypedRedeem(redeem: interbtcIndex.IndexRedeem): Redeem {
     // TODO determine collateralCurrency based on vault
+    const newVaultCollateralAmount = currencyFactory(redeem.collateralCurrencyName) as Factory<CollateralUnit>;
```

Comment:
The currency object should first be retrieved based on the name
```suggestion
    const collateralCurrency = currencyNameToCurrency(redeem.collateralCurrencyName) as Currency<CollateralUnit>;
    const newVaultCollateralAmount = currencyFactory(collateralCurrency);
```